### PR TITLE
Use tick library for time

### LINKED ISSUE #133: Rewrite time built-in functions to use the tick library
The [tick](https://github.com/juxt/tick ) library (which is based on the lower-level [cljc.java-time](https://github.com/henryw374/cljc.java-time)) seems to provide a great abstraction for dealing with time across both Clojure and Clojurescript, and should a clear improvement over what we currently have (which is Clojure only, and rather hacky).

### PR DESCRIPTION
**Pull request by anderseknert**
This provides a unified way of dealing with time for both clj and cljs. As an added bonus, this allowed removing some node-specific helpers as well.

Resolves #133

Signed-off-by: Anders Eknert <anders@eknert.com>

### CODE REVIEW

File: core/src/main/cljc/jarl/utils.cljc [hunk 1]
```diff
@@ -32,19 +32,17 @@
       (update m k indiscriminate-assoc-in ks v)
       (assoc m k v))))
 
-#?(:clj (defn instant-to-ns [^Instant instant]
-          (+' (*' (.getEpochSecond instant) 1000000000) (.getNano instant))))
+(defn instant->ns [instant]
+  #?(:clj  (+' (*' (i/get-epoch-second instant) 1000000000) (t/nanosecond instant))
```
Comment (johanfylling) [inline core/src/main/cljc/jarl/utils.cljc:36]: This difference is for auto-promoting longs? Which isn't supported by `cljs`?

File: core/src/main/cljc/jarl/builtins/time.cljc [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+   "Z070000"   "'Z'XX"
+   "Z07"       "'Z'X"
+   "Z07:00"    "'Z'XXX"
+   "Z07:00:00" "'Z'XXX"
+   "-0700"     "XX"
+   "-070000"   "'Z'XX"
+   "-07"       "X"
+   "-07:00"    "XXX"
+   "-07:00:00" "XXX"
+   "999999999" "SSS"})
```
Comment (johanfylling) [inline core/src/main/cljc/jarl/builtins/time.cljc:46]: Man, I despise this golang time template format .. 

I'd be down for us opening up here, and additionally support the more sane format used internally.

**Review by johanfylling — APPROVED** [review]
👍

File: core/src/test/clj/test/compliance/generator.clj [hunk 3]
```diff
@@ -24,7 +24,17 @@
      "cryptohmacmd5/crypto.hmac.md5_unicode"
      "cryptohmacsha1/crypto.hmac.sha1_unicode"
      "cryptohmacsha256/crypto.hmac.sha256_unicode"
-     "cryptohmacsha512/crypto.hmac.sha512_unicode"}})
+     "cryptohmacsha512/crypto.hmac.sha512_unicode"
+     ; cljs / javascript implementation not limited in this regard — arguably a feature and not a bug
```
[the remark this answers is not in the document]
Comment (anderseknert) [reply core/src/test/clj/test/compliance/generator.clj:28]: Sure! Can add a note on this in the built-in docs 👍

**Pull request merged**